src/App.js

```diff
@@ -1,35 +1,63 @@
-import React, { useEffect, useState } from 'react';
+import React from 'react';
 import { BrowserRouter, Route, Switch, Redirect } from 'react-router-dom';
+import { useCollectionData } from 'react-firebase-hooks/firestore';
+
+import getToken from './lib/tokens';
+import { db } from './lib/firebase';
+import useLocalStorageState from './hooks/useLocalStorageState';
+
+import Home from './components/Home';
 import AddView from './components/AddView';
 import ListView from './components/ListView';
 import Navigation from './components/Navigation';
-import Home from './components/Home'
-import getToken from './lib/tokens';
 import './App.css';
 
 const App = () => {
-  const initialToken = localStorage.getItem('token')
-    ? localStorage.getItem('token')
-    : '';
-
-  const [token, setToken] = useState(initialToken);
+  const [token, setToken] = useLocalStorageState('token', '');
+  const [collectionId, setCollectionId] = useLocalStorageState(
+    'collectionId',
+    [],
+  );
 
-  useEffect(() => {
-    localStorage.setItem('token', token);
-  }, [token]);
+  const addListItem = async (token) => {
+    return await db.collection('lists').add({
+      items: [],
+      token,
+    });
+  };
 
-  const handleTokenCreation = () => {
-    setToken(getToken());
+  const handleTokenCreation = async () => {
+    const token = getToken();
+    setToken(token);
+    const { id } = await addListItem(token);
+    setCollectionId(id);
   };
+
+  const [shoppingList, loading, error] = useCollectionData(
+    db.collection('lists').where('token', '==', token),
```

Comment:
Not sure I understand what this bit of code is doing, but shouldn't the collection be `shoppinglist` instead of `lists`?

Reply:
Hi @C-Zekeri , you're right that's how it was named before. But it would probably be more logical to have  ``shoppinglist`` changed to ``lists`` - because in the app, we will have a bunch of users creating a bunch of lists.

Reply:
If it's about naming, you could just edit the name of the collection in the database, instead of writing to a new collection, yeah?